test/testcafe/spec/DeviceChallengePollView_spec.js

```diff
@@ -317,6 +405,32 @@ test
     await t.expect(identityPage.getIdentifierValue()).eql('Test Identifier');
   });
 
+test
+  .requestHooks(loopbackPollMockLogger, loopbackPollFailedMock)('next poll should not start if previous is failed', async t => {
+    loopbackPollMockLogger.clear();
+    await setup(t);
+    await wait(10_000);
+
+    await t.expect(loopbackPollMockLogger.count(
+      record => record.request.url.match(/\/idp\/idx\/authenticators\/poll/)
+    )).eql(1);
+  });
+
+test
+  .requestHooks(loopbackPollMockLogger, loopbackPollTimeoutMock)('new poll does not starts until last one is ended', async t => {
+    loopbackPollMockLogger.clear();
+    await setup(t);
+    // This test verify if new /poll calls are made only if the previous one was finished instead of polling with fixed interval.
+    // Updating /poll response to take 5 sec to response.
+    // Then counting the number of calls that should be done in time interval. Default Timeout for /poll is 2 sec.
+    // Expecting to get only 2 calls(first at 2nd sec, second at 9th(5 sec response + 2 sec timeout) second).
+    await wait(10_000);
```

Comment:
I may want to write test in such way
```
wait 2 second
expect poll called once
wait another 3 second
expect poll called once
wait another 10 second
expect poll called twice
```

Reply:
As per our conversation added a new [ticket](https://oktainc.atlassian.net/browse/OKTA-465711)
I have played about this approach a little bit and it looks like this one will be tricky(as always when you need to test something with timeline), so it is good that we decided to go with separate ticket